File: src - c++/Esercizi/Somma/somma.cpp
//Metodo privato

void AlberoBinario::_Somma(const PNodo& n, int i) {
  if (n && (i != 0)) {
    n->el += i;
    _Somma(n->sin, i);
    _Somma(n->des, i);
  }  
}

//Metodo Pubblico

void AlberoBinario::Somma(int i) {
  _Somma(root, i);
}

File: src - java/Esercizi/Somma/Somma.java
	private void _somma(Nodo n, int i) {
		if (n != null) {
			n.elem += i;
			_somma(n.sin, i);
			_somma(n.des, i);
		}
	}

	public void somma(int i) {
		if (i != 0)
			_somma(root, i);
	}